Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SpatialCoordinateSystemParticipant.cs

```diff
@@ -6,267 +6,359 @@
-        public void ReceiveMessage(BinaryReader reader)
-        {
-            lock (cancellationTokenSource)
-            {
-                processIncomingMessages?.Invoke(reader);
-            }
-        }
+        public bool PeerDeviceHasTracking { get; internal set; }
 
-        public bool TryTransformToHostWorldSpace(Vector3 localWorldPosition, out Vector3 hostWorldPosition)
-        {
-            hostWorldPosition = Vector3.zero;
-            if (!validLocalOrigin || !validRemoteOrigin)
-            {
-                return false;
-            }
+        /// <summary>
+        /// Gets the last-reported status of whether or not the peer's spatial coordinate is located and tracking.
+        /// </summary>
+        public bool PeerSpatialCoordinateIsLocated { get; internal set; }
 
-            if (isHost)
-            {
-                hostWorldPosition = localWorldPosition;
-            }
-            else
-            {
-                DebugLog($"Local Origin Position: {localOriginPositionInCoordinateSystem.ToString("G4")}, Remote Origin Position: {remoteOriginPositionInCoordinateSystem.ToString("G4")}");
-                hostWorldPosition = localWorldPosition + localOriginPositionInCoordinateSystem - remoteOriginPositionInCoordinateSystem;
-            }
+        /// <summary>
+        /// Gets whether or not the peer device is actively attempting to locate the shared spatial coordinate.
+        /// </summary>
+        public bool PeerIsLocatingSpatialCoordinate { get; internal set; }
 
-            return true;
-        }
+        /// <summary>
+        /// Gets the position of the shared spatial coordinate in the peer device's world space.
+        /// </summary>
+        public Vector3 PeerSpatialCoordinateWorldPosition { get; internal set; }
 
-        public bool TryTransformToHostWorldSpace(Quaternion localWorldRotation, out Quaternion hostWorldRotation)
+        /// <summary>
+        /// Gets the rotation of the shared spatial coordinate in the peer device's world space.
+        /// </summary>
+        public Quaternion PeerSpatialCoordinateWorldRotation { get; internal set; }
+
+        public string PersistentCoordinateId { get; internal set; }
+
+        private void Update()
         {
-            hostWorldRotation = Quaternion.identity;
-            if (!validLocalOrigin || !validRemoteOrigin)
+            if (Coordinate != null && Coordinate.State == LocatedState.Tracking)
             {
-                return false;
+                this.transform.position = Coordinate.CoordinateToWorldSpace(Vector3.zero);
+                this.transform.rotation = Coordinate.CoordinateToWorldSpace(Quaternion.identity);
             }
 
-            if (isHost)
+            if (SocketEndpoint != null && SocketEndpoint.IsConnected)
             {
-                hostWorldRotation = localWorldRotation;
+                SendCoordinateStateMessage();
             }
```

Comment:
do we need to send this every frame? For keynote monitoring, i get that you may want to know the state of things. but for mobile experiences, this may be added noise on the main user's end when having multiple viewers.

Reply:
Just a key-frame animator's two cents.  Would be nice if eventually it was sent every frame but for most functionality i think once every second or even every other second would be sufficient.  There will be various forms of Lag anyway.  This issue has been on my mind for weeks and I would hate to see the need for perfect getting in the way of it functioning correctly.  I am assuming very subtle delays will be necessary.

Reply:
The state message is checked every frame, but a message is only sent if the status ends up being different from the previous frame. This should only happen when:
1. Tracking is lost or gained
2. The WorldAnchor adjusts the position of a WorldAnchor-based coordinate
3. A new marker detection session starts or ends

If there's noise from this message type, we should figure out what's changing on a per-frame basis and try to alleviate it.